src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int CAPACITY_INCREASE = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table = new Node[DEFAULT_CAPACITY];
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        checkThreshold();
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int bucketIndex = getBucketIndex(hash(key));
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[bucketIndex] == null) {
+            table[bucketIndex] = newNode;
+        } else {
+            connectNode(newNode, bucketIndex);
+            return;
+        }
+        Node<K, V> currentNode = table[bucketIndex];
+        if (currentNode.key.equals(key)) {
+            currentNode.value = value;
+        } else {
+            currentNode.next = newNode;
```

Comment:
I guess you don`t need to check separately for null key, try to find a method that will have universal approach